xonsh/pyghooks.py

```diff
@@ -21,11 +21,17 @@
 from pygments.styles import get_style_by_name
 import pygments.util
 
+from xonsh.commands_cache import CommandsCache
 from xonsh.lazyasd import LazyObject, LazyDict
 from xonsh.tools import (ON_WINDOWS, intensify_colors_for_cmd_exe,
                          expand_gray_colors_for_cmd_exe)
-from xonsh.tokenize import SearchPath
-
+                         
+# If the lexor is loaded as a pygment plugin, we have to mock  
+# __xonsh_env__ and __xonsh_commands_cache__ 
+if not hasattr(builtins, '__xonsh_commands_cache__'):
+    if not hasattr(builtins, '__xonsh_env__'):
+        setattr(builtins, '__xonsh_env__', {})
```

Comment:
No. That would be a detyped environment. What the `command_cache` need is access to a few entries in the xonsh form.